doc/pandas-2.0/source/strings.rst

```diff
@@ -0,0 +1,195 @@
+.. _strings:
+
+.. ipython:: python
+   :suppress:
+
+   import numpy as np
+   import pandas as pd
+   np.set_printoptions(precision=4, suppress=True)
+   pd.options.display.max_rows = 100
+
+==================================
+ Enhanced string / UTF-8 handling
+==================================
+
+There are some things we can do to make pandas use less memory and perform
+computations significantly faster on string data.
+
+Current string problems
+=======================
+
+pandas offers support for columns containing strings (ASCII or Unicode) on a
+somewhat ad hoc basis.
+
+* Strings are stored in NumPy arrays of ``PyObject*`` / ``numpy.object_``
+  dtype. This has several problems
+
+  * Computations (e.g. ``groupby`` operations) typically utilize a code path
+    for generic Python objects. For example comparisons or hashing goes through
+    the ``PyObject_*`` C API functions. In addition to harming multithreading
+    due to GIL contention (you must acquire the GIL to use these functions),
+    these can also be significantly slower than algorithms that operate on
+    ``const char*``, potentially taking advantage of hardware optimizations.
+
+  * String arrays often feature many copies of or references to the same
+    PyString. Thus, some algorithms may perform redundant computation. Some
+    parts of pandas, like ``pandas.read_csv``, make an effort to deduplicate
+    strings to free memory and accelerate computations (e.g. if you do ``x ==
+    y``, and ``x`` and ``y`` are references to the same ``PyObject*``, Python
+    skips comparing their internal data).
+
+    * Note that this is somewhat mitigated by using ``pandas.Categorical``, but
+      this is not the default storage mechanism. More on this below.
+
+  * Using ``PyString`` objects and ``PyObject*`` NumPy storage adds non-trivial
+    overhead (approximately 24 bytes per unique object, see `this exposition
+    <http://www.gahcep.com/python-internals-pyobject/>`_ for a deeper drive) to
+    each value.
+
+Possible solution: new non-NumPy string memory layout
+=====================================================
+
+My proposed solution to the string conundrum is the following:
+
+* Create a custom string array container type suitable for use in a
+  ``pandas.Array``, and a ``pandas.string`` logical data type.
+* Require that all strings be encoded as UTF-8.
```

Comment:
My understanding is that "Unicode" is a generic term, versus UTF-8 refers to a specific encoding. Python's Unicode type uses UTF-8 internally, but also some other metadata (type, reference counter plus reference to the data -- this is the 24 bytes of PyObject overhead) that means creating it requires memory allocation.